### PR TITLE
Limit comment message to 1k chars

### PR DESCRIPTION
**Pull request by blizzz**
Enforces that a provided message length is no longer than 1k characters. Backend companion to https://github.com/owncloud/core/pull/22163#issuecomment-180431700

Please review @PVince81 @icewind1991 @nickvergessen

### CODE REVIEW

File: lib/public/comments/messagetoolongexception.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+namespace OCP\Comments;
+
+/**
+ * Exception for not found entity
```
Comment (LukasReschke) [inline lib/public/comments/messagetoolongexception.php]: ;)

File: lib/private/comments/comment.php [hunk 2]
```diff
@@ -184,13 +185,18 @@ public function getMessage() {
...
 	 * @since 9.0.0
 	 */
 	public function setMessage($message) {
 		if(!is_string($message)) {
 			throw new \InvalidArgumentException('String expected.');
 		}
-		$this->data['message'] = trim($message);
+		$message = trim($message);
+		if(mb_strlen($message, 'UTF-8') > IComment::MAX_MESSAGE_LENGTH) {
+			throw new MessageTooLongException('Comment message must not exceed ' . IComment::MAX_MESSAGE_LENGTH . ' characters');
```
Comment (PVince81) [inline lib/private/comments/comment.php]: Make sure this is translated to a BadRequest sabre exception in the relevant handlers. Else you'll get an ugly 500 Internal Server Error.

File: lib/public/comments/messagetoolongexception.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * @since 9.0.0
+ */
+class MessageTooLongException extends \Exception {}
```
Comment (nickvergessen) [inline lib/public/comments/messagetoolongexception.php]: Could extend `\OverflowException`

File: lib/private/comments/comment.php (hunk 2 above)
Comment (nickvergessen) [inline lib/private/comments/comment.php]: +1 for this

**Comment by blizzz** [comment]
PHP doc and dav responsens fixed

**Comment by PVince81** [comment]
:+1:

**Comment by PVince81** [comment]
Second review @nickvergessen @rullzer @icewind1991 ?

**Comment by rullzer** [comment]
Works as expected.
Code looks clean :) :+1:

**Pull request merged**